### PR TITLE
[go/program-gen] Implement importPathPattern in Go language options to override emitted paths in generated Go programs

### PR DESCRIPTION
**Pull request by Zaid-Ajaj**
# Description

Partially addressing https://github.com/pulumi/pulumi-azure-native/issues/3308 

Implementing a new go language option called `importPathPattern` which can be used to override how the base import path and modules are concatenated to create imports in generated Go programs. By convention this used to be `{baseImportPath}/{module}` which has worked for all of our providers. However, azure-native v2 has introduced a new import scheme where the convention above causes incorrect import paths to be generated.

This is where `importPathPattern` comes into play and allows for a different convention. In the case the of azure-native v2, the pattern _must_ be `github.com/pulumi/pulumi-azure-native-sdk/{module}/v2`. 

This PR implements `importPathPattern` and tests it using a squashed down azure-native v2 schema containing only contents from the `eventgrid` module. This schema sets the option like this:
```json
"importPathPattern": "github.com/pulumi/pulumi-azure-native-sdk/{module}/v2"
```
This schema also modifies `packageImportAliases` from the current azure-native v2 schema to exclude /v2 before the module path (see the file below). This change is needed in the actual azure-native v2 provider cc @danielrbradley and it is the second part of fully fixing https://github.com/pulumi/pulumi-azure-native/issues/3308

> We cannot just use `{baseImportPath}/{module}` because the base import path for azure-native v2 has a suffix v2 (it's required) 

Also implemented a small feature in ProgramTest to allow overriding the used plugin host for the specific program test, this is because I wanted to test the program against the simplified azure-native v2 schema without changing how other test load the previous azure-native v1.x schemas (I tested that without this, binding programs fails)

## Checklist

- [ ] I have run `make tidy` to update any new dependencies
- [x] I have run `make lint` to verify my code passes the lint check
  - [x] I have formatted my code using `gofumpt`

- [ ] I have added tests that prove my fix is effective or that my feature works

- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

**Review by danielrbradley — COMMENTED** [review]
Looks good!

**Review by danielrbradley — APPROVED** [review]

File: tests/testdata/codegen/azure-native-2.41.0.json
Comment (tgummerer) [inline tests/testdata/codegen/azure-native-2.41.0.json:1]: Do we need to import this whole schema file here? Would it be possible to have something smaller that just contains the schema bits we need?

**Review by tgummerer — APPROVED** [review]
The code changes look reasonable to me.  I was wondering if it's possible to not have the whole schema file checked in, since it's quite large, and kinda hard to see which parts are needed for the test and which aren't.

File: tests/testdata/codegen/azure-native-2.41.0.json
Comment (Zaid-Ajaj) [reply tests/testdata/codegen/azure-native-2.41.0.json:1]: This schema file is already the squashed down version from the full azure-native schema. I only took a single module from there (the one called `eventgrid`) and left out everything else. It is possible to slim it even further and not everything is needed for this particular feature but I am working on more fixes for azure native and will be needing more things from this same schema.

Comment (Zaid-Ajaj) [reply tests/testdata/codegen/azure-native-2.41.0.json:1]: Created an issue https://github.com/pulumi/pulumi/issues/16280 to follow this up in a later PR.

**Pull request merged**